lib/kazoo/partition.rb

```diff
@@ -16,7 +16,7 @@ def replication_factor
     end
 
     def preferred_leader
-      @replicas.first
+      @replicas.compact.first
```

Comment:
My preference would be that we still have a `Broker` object with the proper ID, but with an `unavailable?` method that will return true in that case. That way we can do things like `partition.preferred_replica.available?`